pallets/teebag/src/lib.rs

```diff
@@ -0,0 +1,833 @@
+			Ok(().into())
+		}
+
+		#[pallet::call_index(2)]
+		#[pallet::weight((195_000_000, DispatchClass::Normal, Pays::No))]
+		pub fn set_max_enclave_count(
+			origin: OriginFor<T>,
+			worker_type: WorkerType,
+			new_count: u64,
+		) -> DispatchResultWithPostInfo {
+			Self::ensure_admin_or_root(origin)?;
+			ensure!(
+				new_count > Self::max_enclave_count(worker_type),
+				Error::<T>::MaxEnclaveCountUnderflow
+			);
+
+			MaxEnclaveCount::<T>::insert(worker_type, new_count);
+			Self::deposit_event(Event::MaxEnclaveCountSet { worker_type, new_count });
+			Ok(Pays::No.into())
+		}
+
+		#[pallet::call_index(3)]
+		#[pallet::weight((195_000_000, DispatchClass::Normal, Pays::No))]
+		pub fn force_add_enclave(
+			origin: OriginFor<T>,
+			who: T::AccountId,
+			enclave: Enclave,
+		) -> DispatchResultWithPostInfo {
+			Self::ensure_admin_or_root(origin)?;
+			Self::add_enclave(&who, &enclave)?;
+			Ok(Pays::No.into())
+		}
+
+		#[pallet::call_index(4)]
+		#[pallet::weight((195_000_000, DispatchClass::Normal, Pays::No))]
+		pub fn force_remove_enclave(
+			origin: OriginFor<T>,
+			who: T::AccountId,
+		) -> DispatchResultWithPostInfo {
+			Self::ensure_admin_or_root(origin)?;
+			Self::remove_enclave(&who)?;
+			Ok(Pays::No.into())
+		}
+
+		#[pallet::call_index(5)]
+		#[pallet::weight((195_000_000, DispatchClass::Normal, Pays::No))]
+		pub fn force_remove_enclave_by_mrenclave(
+			origin: OriginFor<T>,
+			mrenclave: MrEnclave,
+		) -> DispatchResultWithPostInfo {
+			Self::ensure_admin_or_root(origin)?;
+			let accounts = EnclaveRegistry::<T>::iter()
+				.filter_map(
+					|(who, enclave)| {
+						if enclave.mrenclave == mrenclave {
+							Some(who)
+						} else {
+							None
+						}
+					},
+				)
+				.collect::<Vec<T::AccountId>>();
+
+			for who in accounts.into_iter() {
+				Self::remove_enclave(&who)?;
+			}
+			Ok(Pays::No.into())
+		}
+
+		#[pallet::call_index(6)]
+		#[pallet::weight((195_000_000, DispatchClass::Normal, Pays::No))]
+		pub fn force_remove_enclave_by_worker_type(
+			origin: OriginFor<T>,
+			worker_type: WorkerType,
+		) -> DispatchResultWithPostInfo {
+			Self::ensure_admin_or_root(origin)?;
+			let accounts = EnclaveRegistry::<T>::iter()
+				.filter_map(
+					|(who, enclave)| {
+						if enclave.worker_type == worker_type {
+							Some(who)
+						} else {
+							None
+						}
+					},
+				)
+				.collect::<Vec<T::AccountId>>();
+
+			for who in accounts.into_iter() {
+				Self::remove_enclave(&who)?;
+			}
+
+			Ok(Pays::No.into())
+		}
+
+		#[pallet::call_index(7)]
+		#[pallet::weight((195_000_000, DispatchClass::Normal, Pays::No))]
+		pub fn set_scheduled_enclave(
+			origin: OriginFor<T>,
+			worker_type: WorkerType,
+			sidechain_block_number: SidechainBlockNumber,
+			mrenclave: MrEnclave,
+		) -> DispatchResultWithPostInfo {
+			Self::ensure_admin_or_root(origin)?;
+			ensure!(worker_type.is_sidechain(), Error::<T>::ScheduledEnclaveNotApplicable,);
```

Comment:
I thought this should allow to set mrenclave for other worker types too.

	// For NON-sidechain worker_type, we still use this storage to whitelist mrenclave, in this case
	// the `SidechainBlockNumber` is ignored - you could always set it to 0.

Reply:
Updated!